build/release-tars/BUILD

```diff
@@ -0,0 +1,244 @@
+)
+
+pkg_tar(
+    name = "kubernetes-src",
+    extension = "tar.gz",
+    files = [
+        "//:all-srcs",
+    ],
+    package_dir = "kubernetes",
+    strip_prefix = "//",
+)
+
+# FIXME: this should be configurable/auto-detected
+PLATFORM_ARCH_STRING = "linux-amd64"
+
+# KUBE_CLIENT_TARGETS
+CLIENT_TARGETS = [
+    "//cmd/kubectl",
+    "//federation/cmd/kubefed",
+]
+
+# KUBE_NODE_TARGETS
+NODE_TARGETS = [
+    "//cmd/kube-proxy",
+    "//cmd/kubelet",
+]
+
+# KUBE_SERVER_TARGETS
+# No need to duplicate CLIENT_TARGETS or NODE_TARGETS here,
+# since we include them in the actual build rule.
+SERVER_TARGETS = [
+    "//cmd/hyperkube",
+    "//cmd/kube-aggregator",
+    "//cmd/kube-apiserver",
+    "//cmd/kube-controller-manager",
+    "//cmd/kube-discovery",
+    "//cmd/kubeadm",
+    "//plugin/cmd/kube-scheduler",
+]
+
+# kube::golang::test_targets
+TEST_BINARY_TARGETS = [
+    "//cmd/gendocs",
+    "//cmd/genkubedocs",
+    "//cmd/genman",
+    "//cmd/genswaggertypedocs",
+    "//cmd/genyaml",
+    "//cmd/linkcheck",
+    "//cmd/mungedocs",
+    "//examples/k8petstore/web-server/src",
+    "//federation/cmd/genfeddocs",
+    "//test/e2e:e2e.test",
+    "//vendor:github.com/onsi/ginkgo/ginkgo_bin",
+    "//cmd/kubemark",  # TODO: server platforms only
+    "//test/e2e_node:e2e_node.test",  # TODO: server platforms only
+]
+
+TEST_PORTABLE_TARGETS = [
+    "//federation/develop:all-srcs",
+    "//hack:e2e.go",
+    "//hack:federated-ginkgo-e2e.sh",
+    "//hack:get-build.sh",
+    "//hack:ginkgo-e2e.sh",
+    "//hack/e2e-internal:all-srcs",
+    "//hack/lib:all-srcs",
+    "//test/e2e/testing-manifests:all-srcs",
+    "//test/kubemark:all-srcs",
+]
+
+# Included in node and server tarballs.
+LICENSE_TARGETS = [
+    "//:Godeps/LICENSES",
+    ":kubernetes-src.tar.gz",
+]
+
+pkg_tar(
+    name = "_client-bin",
+    files = CLIENT_TARGETS,
+    mode = "0755",
+    package_dir = "client/bin",
+    visibility = ["//visibility:private"],
+)
+
+pkg_tar(
+    name = "kubernetes-client-%s" % PLATFORM_ARCH_STRING,
+    extension = "tar.gz",
+    package_dir = "kubernetes",
+    deps = [
+        ":_client-bin",
+    ],
+)
+
+pkg_tar(
+    name = "_node-bin",
+    files = NODE_TARGETS + CLIENT_TARGETS,
+    mode = "0755",
+    package_dir = "node/bin",
+    visibility = ["//visibility:private"],
+)
+
+pkg_tar(
+    name = "kubernetes-node-%s" % PLATFORM_ARCH_STRING,
+    extension = "tar.gz",
+    files = LICENSE_TARGETS,
+    mode = "0644",
+    package_dir = "kubernetes",
+    deps = [
+        ":_node-bin",
+    ],
+)
+
+pkg_tar(
+    name = "_server-bin",
+    files = SERVER_TARGETS + NODE_TARGETS + CLIENT_TARGETS + [
+        "//build:docker-artifacts",
+    ],
+    mode = "0755",
+    package_dir = "server/bin",
+    visibility = ["//visibility:private"],
+)
+
+genrule(
+    name = "dummy",
+    outs = [".dummy"],
+    cmd = "touch $@",
+)
+
+# Some of the startup scripts fail if there isn't an addons/ directory in the server tarball.
```

Comment:
yeah, this part makes me sad. I'm not sure how many places assume `server/addons/` exists, but that can hopefully be addressed in a follow-up.